src/files/domain/useCases/addUploadedFiles.ts

```diff
@@ -14,18 +14,3 @@ export function addUploadedFiles(
     })
     .finally(done)
```

Comment:
The `done` function is being called in the `.finally` block here, but that is not a guarantee that the use case ran correctly,  the finally block just tells you that the promise is settled regardless of whether it was successful or not.
Instead, try with this:

```
  fileRepository
    .addUploadedFiles(datasetId, files)
    .then(done)
    .catch((error: Error) => {
      throw new Error(error.message)
    })
```